docs/proposals/going-towards-to-crd.md

```diff
@@ -0,0 +1,104 @@
+# Service Catalog going towards to CRDs
+
+ In Service Catalog we want to replace the Aggregated API with the CRD solution. This document specifies the concerns and possible solutions for supporting Service Catalog with an Aggregated API and the new Custom Resource Definitions (CRDs) approach.
+
+## Support both Aggregated API and CRDs in the same code-base
+
+Below you will find our concerns about having a single executable binary that supports both the Aggregated API and CRDs in the same code-base and changing the behavior by using a feature gate. 
+
+### Business concerns
+
+Adding CRDs via feature flag directly in Helm Chart can be misleading for the client. 
+The CRDs it’s not a feature. It is just a new implementation for already existing feature. Leaving configuration [`apiserver.storage.type`: crds/etcd](https://github.com/kubernetes-incubator/service-catalog/blob/master/charts/catalog/values.yaml#L61-L62) is more like extending but not deprecating it. We do not want to make that customer think that **CRD** or **etcd** is an option as storage. Customers should know that the etcd is deprecated and they should consider switching for CRDs approach as soon as possible.  
+
+Additionally, supporting bugs fixing and adding new features both in Aggregated API and CRDs could slow-down the development process.    
+
+Another problem is to have consistent and up-to-date documentation. When we will support both approaches then we need to maintain documentation for them too. One more time it could be misleading for customers if we will show them that we are going in two directions at the same time.    
+
+### Technical concerns
+
+We need to clearly state that in Service Catalog the Aggregated API and the CRDs are not only about the underlying storage backend. Around those approaches, we have business logic. Because of that, we will end up with a lot of `if` statements in:
```

Comment:
```suggestion
We need to clearly state that in the Service Catalog, the Aggregated API and the CRDs are not only about the underlying storage backend. Around those approaches, we have business logic. Because of that, we will end up with a lot of `if` statements in:
```